Wikipedia/Code/WMFAppViewController.m

```diff
@@ -207,14 +220,74 @@ - (void)pauseApp {
     [self performHousekeepingIfNecessary];
 }
 
+- (void)handleSelectionOfShortcutItemIfNecessaryThen:(void (^)(BOOL shortcutWasHandled))completion {
+    UIApplicationShortcutItem* shortcutItemSelectedAtLaunch = self.shortcutItemSelectedAtLaunch;
+    if (shortcutItemSelectedAtLaunch) {
+        // Main UI may *not* have been loaded if the user...
+        //      - minimized the app when viewing a "Welcome" screen
+        //      - then selected a 3d touch app icon shortcut menu item
```

Comment:
If the user hasn't finished viewing the welcome screen, should there be any actions in the short cut menu?
Users probably shouldn't be able to use the shortcut menu to "skip" the welcome screen.